lib/tls/session.toit

```diff
@@ -180,6 +213,18 @@ class Session:
       throw "TLS_ALREADY_HANDSHAKEN"
 
     handshake_in_progress_ = monitor.Latch
+
+    if session_state:
+      toit_handshake_ = ToitHandshake_ this
+      try:
+        toit_handshake_.handshake
```

Comment:
Do you need to keep the `toit_handshake_` around after this? Seems a bit off. Maybe it shouldn't be stored in a field at all.